two-regions-app-lb/region.yaml

```diff
@@ -0,0 +1,132 @@
+heat_template_version: 2016-04-08
+
+description: a single server backend
+
+parameters:
+
+  app_port:
+    type: number
+  public_network:
+    type: string
+    description: Network used by instances
+    default: ext-net
+  image:
+    type: string
+  ssh_keys:
+    type: comma_delimited_list
+  backend_name: 
+    type: string
+    description: Name of this backend target
+  
+
+resources:
+
+  allow_ssh:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      description: allow incoming SSH and ICMP traffic from anywhere.
+      name: allow incoming traffic, tcp port 22 and icmp
+      rules: 
+        - { direction: ingress, remote_ip_prefix: 0.0.0.0/0, port_range_min: 22, port_range_max: 22, protocol: tcp } 
+        - { direction: ingress, remote_ip_prefix: 0.0.0.0/0, protocol: icmp } 
+
+  allow_webtraffic:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      description: allow incoming web traffic via http from anywhere. 
+      name: allow incoming traffic, tcp port 80
+      rules: 
+        - { direction: ingress, remote_ip_prefix: 0.0.0.0/0, port_range_min: { get_param: app_port } , port_range_max: { get_param: app_port } , protocol: tcp } 
+
+  host:
+    type: OS::Nova::Server
+    properties:
+      name: { get_param: backend_name }
+      user_data_format: RAW
+      user_data: { get_resource: cloud-init-config }    
+      image: { get_param: image }
+      flavor: m1.tiny
+      networks:
+        - port: { get_resource: port }
+
+  cloud-init-config:
+   # cloud init resource
+   type: OS::Heat::CloudConfig
+   properties:
+     cloud_config:
+       manage_etc_hosts: true
+       users:
+           - name: syseleven
+             gecos: SysEleven Stack user
+             sudo: ALL=(ALL) NOPASSWD:ALL
+             shell: /bin/bash
+             ssh-authorized-keys: { get_param: ssh_keys }
+       write_files:
+         -  content: { get_file: scripts/install_lampstack.sh }
+            permissions: 700
+            path: /root/install_lampstack.sh
+         -  content: |
+              #!/usr/bin/env bash
+              # get AnyApp php
+              curl "https://raw.githubusercontent.com/syseleven/heat-examples/master/sources/AnyApp/index.php" > /var/www/html/index.php
+              chown www-data:www-data /var/www/html/index.php
+              rm -f /var/www/html/index.html
+            path: /root/inject_anyapp
+            permissions: 0700
+       runcmd:
+         - /root/install_lampstack.sh
+         - /root/inject_anyapp
```

Comment:
When using anyapp and anyapp uses sessions to build the server list then we should probably configure sessions. Otherwise we could leave the static page. A list containing the current backend only does not look good.

Reply:
We can update the anyapp to only show the "current backend nodes" if there is more than one.

Reply:
That would be a cosmetic workaround but not a solution. Users of our blocks might want to know whether they can expect session handling to work in a building block or not. An application that on the one hand uses them but on the other hand doesn't really need them is not helpful. I'd rather make different versions or just make the static html page looking better.